### PR TITLE
[skip ci] add <Leader>c macro for appending [skip ci] (see this commit message for more info). :)

### PR DESCRIPTION
**Pull request by treznick**
For :tada: and :dollar:!

### CODE REVIEW

**Comment by benjaminoakes** [comment]
Feels like it might be better to add this after the first line

```
Commit message [#000000]

[skip ci]
```

Otherwise it's noise at the beginning of the line, which is no fun on GitHub.  Prevents you from seeing useful info.

**Comment by treznick** [comment]
maybe append to the end instead? Is pivotal's tracking position sensitive?

**Comment by treznick** [comment]
per @benjaminoakes and @danbernier 's comments, this now adds below the line, and it's now `<Leader>skip`

**Comment by treznick** [comment]
also forgot to unhitch @caseywatts , who surely has better things to do at 11:30 on a tuesday...

File: home/.vimrc [hunk 1]
```diff
@@ -97,7 +97,7 @@ map \ :NERDTreeToggle<CR>
 
 map <Leader>p gg:s/\s*$<CR>/\d\+-<CR>ywggA [#<esc>pa]<esc>
 nmap <Leader>l iLorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.<esc>
-
+nmap <Leader>skip gg2o<esc>i[skip ci]<esc>
```
Comment (benjaminoakes) [inline home/.vimrc:100]: Why'd this change from `<Leader>c`?  This is long enough I might just forget and just `o<cr>[skip ci]` by hand...

Comment (danbernier) [inline home/.vimrc:100]: I had suggested that we don't rush to claim all the single-char <Leader> triggers. For example, `<Leader>l` for generating lorem-ipsum is pretty useless - it would be better mapped to (say) introduce a `let` in a spec.

We got to talking about it, and how, with `<Leader>skip` we're not saving much. I think @treznick will be looking into making it a snippet.

**Comment by treznick** [comment]
Looking into this being a snippet.

I've given the `.vim` directory a quick gander. Doesn't look like we have any custom snippets (or really any snippets) defined anywhere, and there's nothing in the `runtimepath` that suggests anything otherwise. My gut says that `.vim/bundle/snippets/` should be the place, but I wanted to check with you neckbeardy folks first @benjaminoakes @danbernier @jhirbour @jnimety @tlball

**Comment by jhirbour** [comment]
Relevant: http://code.tutsplus.com/tutorials/vim-essential-plugin-snipmate--net-19356

**Comment by jnimety** [comment]
https://github.com/garbas/vim-snipmate... They reference
https://github.com/honza/vim-snippets in their install instructions which
is not part of our repo

On Wed, Mar 11, 2015 at 1:50 PM, John Hirbour notifications@github.com
wrote:

> Relevant:
> http://code.tutsplus.com/tutorials/vim-essential-plugin-snipmate--net-19356
> 
> ## 
> 
> Reply to this email directly or view it on GitHub
> https://github.com/ContinuityControl/dotfiles/pull/4#issuecomment-78317487
> .

## 

Joel Nimety
Senior VP, Engineering
Continuity
203.285.8133 (office)
203.763.9263 (cell)
http://www.continuity.net
http://www.linkedin.com/in/jnimety

## 

## 

---

This email and all files transmitted with it are intended solely for use by 
the intended recipient.  Any other use is prohibited.  If you received this 
email in error, please notify the sender, delete the email and destroy all 
attachments.  Opinions and ideas expressed herein are solely those of the 
sender, and may not reflect the official position of Continuity.

File: home/.vimrc (hunk 1 above)
Comment (benjaminoakes) [inline home/.vimrc:100]: I like the snippet idea.

I guess I use lorem ipsum a lot more than you, too.  :smile:  Though I have no problem if we decide to snippetize that too.

Comment (treznick) [inline home/.vimrc:100]: More general point: Should vim-snippets be part of our dotfiles? I think it
needs some python on the backend, so not sure what mad puppetry that might
involve.

Barring the addition of vim-snippets, kosher to add a snippets dir to our
vim bundle directory?
-t

2015-03-12 10:52 GMT-04:00 Ben Oakes notifications@github.com:

> In home/.vimrc
> https://github.com/ContinuityControl/dotfiles/pull/4#discussion_r26307920
> :
> 
> > @@ -97,7 +97,7 @@ map \ :NERDTreeToggle<CR>
> > 
> >  map <Leader>p gg:s/\s*$<CR>/\d+-<CR>ywggA [#<esc>pa]<esc>
> > 
> > ##  nmap <Leader>l iLorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.<esc>
> > 
> > +nmap <Leader>skip gg2o<esc>i[skip ci]<esc>
> 
> I like the snippet idea.
> 
> I guess I use lorem ipsum a lot more than you, too. [image: :smile:]
> Though I have no problem if we decide to snippetize that too.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ContinuityControl/dotfiles/pull/4/files#r26307920.

## 

Tom Reznick
Junior Software Engineer, Continuity
914-519-8966

## 

## 

---

This email and all files transmitted with it are intended solely for use by 
the intended recipient.  Any other use is prohibited.  If you received this 
email in error, please notify the sender, delete the email and destroy all 
attachments.  Opinions and ideas expressed herein are solely those of the 
sender, and may not reflect the official position of Continuity.

Comment (benjaminoakes) [inline home/.vimrc:100]: We have Python 2.7.x; part of base Fedora I believe.

:+1: to having a snippets dir

**Comment by treznick** [comment]
tl;dr: Tried adding vim-snippets to our repo, but it causes conflicts with supertab

We've already got [snipmate](https://github.com/garbas/vim-snipmate), but since we have no snippets `<tab>` in insert mode defaults to hitting [supertab](https://github.com/ervandew/supertab)'s tab completion functionality.

When I installed [vim-snippets](https://github.com/honza/vim-snippets), this got all wonky, such that if there's a tab completion that overrides what you're trying to tab on currently (e.g. hitting `<tab>` after `def` puts you into a create method snippet in ruby, instead of say, trying to complete the word `deferential`).

Also it seems fairly complicated to try to get snippets working for the `gitcommit` filetype.

I'll hack on this during OKR time if anyone more vim-fluent wants to help, but this might be a bit of a rabbit hole for a trivial gain.

Some perfunctory googling suggests that [neocomplcache](https://github.com/Shougo/neocomplcache.vim) with [neosnippets](https://github.com/Shougo/neosnippet.vim) might work, or possibly [Utilisnips](https://github.com/sirver/ultisnips)

**Comment by benjaminoakes** [comment]
I personally don't use tab completion.  The vim way is to do ctrl-p and ctrl-n.  There might be some value in getting people used to that so that we can use snippets.

That said, this does seem to be rabbit holing.  Let's just merge this unless you want to spend more time on it.

**Comment by treznick** [comment]
eh, I'll keep poking at this. It's not mission critical enough to get in regardless, but it's fun enough to solo on here and there

**Comment by jnimety** [comment]
removing snipmate since we don't have any snippets anyway.  I use tab
completion _a ton_ but I'll try ctrl-p/n to see how it compares.  Tom, if
you come up with a solution that uses snipmate feel free to re-add to our
bundle. -- J

On Fri, Mar 13, 2015 at 10:48 AM, Tom Reznick notifications@github.com
wrote:

> eh, I'll keep poking at this. It's not mission critical enough to get in
> regardless, but it's fun enough to solo on here and there
> 
> ## 
> 
> Reply to this email directly or view it on GitHub
> https://github.com/ContinuityControl/dotfiles/pull/4#issuecomment-79007071
> .

## 

Joel Nimety
Senior VP, Engineering
Continuity
203.285.8133 (office)
203.763.9263 (cell)
http://www.continuity.net
http://www.linkedin.com/in/jnimety

## 

## 

---

This email and all files transmitted with it are intended solely for use by 
the intended recipient.  Any other use is prohibited.  If you received this 
email in error, please notify the sender, delete the email and destroy all 
attachments.  Opinions and ideas expressed herein are solely those of the 
sender, and may not reflect the official position of Continuity.

**Comment by treznick** [comment]
will do. closing for now.
